lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -101,12 +102,84 @@ class IntProgCmdProvider: public CmdProvider {
     static const ResponseInfo m_responses[SEQ_LENGHT]; /**< Array of response information */
 };
 
-#if 0
+/**
+ * @class FwProgCmdProvider
+ * @brief Implementation of the command provider for firmware programming.
+ *
+ * This class provides commands for firmware programming, specifically tailored
+ * for the Zumo32U4 device.
+ */
 class FwProgCmdProvider : public CmdProvider {
     public:
-        FwProgCmdProvider(FileHandle * fs);
-}
-#endif
+        /**
+         * @brief Constructor for the FwProgCmdProvider class.
+         * @param fs A pointer to the FileHandler object for handling file operations.
+         */
+        FwProgCmdProvider(FileHandler * fs) :

```

Comment:
Need to rethink what makesn sense as constructor parameter. 
Maybe you need the filename here and open it internally for read instead of getting a handle.

Reply:
t will be taken care of by the next commit.